src/main/kotlin/graphics/scenery/volumes/HistogramChartManager.kt

```diff
@@ -0,0 +1,154 @@
+package graphics.scenery.volumes
+
+import graphics.scenery.SceneryElement
+import org.jfree.chart.ChartPanel
+import org.jfree.chart.axis.LogarithmicAxis
+import org.jfree.chart.axis.NumberAxis
+import org.jfree.chart.plot.XYPlot
+import org.jfree.chart.renderer.xy.StandardXYBarPainter
+import org.jfree.chart.renderer.xy.XYBarRenderer
+import org.jfree.data.statistics.SimpleHistogramBin
+import org.jfree.data.statistics.SimpleHistogramDataset
+import javax.swing.JCheckBox
+import kotlin.math.abs
+import kotlin.math.max
+import kotlin.math.roundToInt
+
+/**
+ * Handles all histogram related things.
+ */
+class HistogramChartManager(val tfPlot: XYPlot,
+                            val mainChart: ChartPanel,
+                            private val tfContainer: HasTransferFunction,
+                            val axisExtensionFactor: Double){
+    val genHistButton = JCheckBox("Show Histogram")
+
+    private val histogramAxis = LogarithmicAxis("")
+
+    init {
+        val histogramRenderer = XYBarRenderer()
+        histogramRenderer.setShadowVisible(false)
+        histogramRenderer.barPainter = StandardXYBarPainter()
+        histogramRenderer.isDrawBarOutline = false
+        tfPlot.setRenderer(1, histogramRenderer)
+
+        val histXAxis = NumberAxis()
+        var range = abs(tfContainer.maxDisplayRange - tfContainer.minDisplayRange)
+        histXAxis.setRange(
+            tfContainer.minDisplayRange - (axisExtensionFactor * range),
+            tfContainer.maxDisplayRange + (axisExtensionFactor * range)
+        )
+
+        histogramAxis.isMinorTickMarksVisible = true
+        histogramAxis.setRange(
+            0.0,
+            1000.0
+        )
+
+        val volumeHistogramData = SimpleHistogramDataset("VolumeBin")
+        volumeHistogramData.adjustForBinSize = false
+
+        if (tfContainer is HasHistogram) {
+            genHistButton.addActionListener() {
```

Comment:
Great that this has become an ActionListener instead of a ChangeListener, I've just worked around that bug in sciview 👍